### ISSUE TITLE
Power Switch Modules

### ISSUE THREAD

**Issue by trymeouteh** [pikvm/pikvm#1445]
**Is your feature request related to a problem? Please describe.**
A simple way to forcefully turn off the computer remotely when the computer freezes, or gets stuck in a loop when shutting down.

**Describe the solution you'd like**
An optional expandable module you can plug into the PiKVM via USB which is a power switch. And will have a male power outlet that plugs into a power source such as a wall outlet and a female power outlet which you will plug your computer into.

Different modules will need to be made for different regions. A US one for US power outlets, an EU one for european outlets, etc

And to have a way in the software to use the module by enabling or disabling the power in the power switch.

**Describe alternatives you've considered**
Using remote power switches such as this one
https://www.amazon.com/Dataprobe-iBoot-G2-Network-Automation-Rebooting/dp/B00B0YJUPQ/

However I was unable to find one that is fully open source.

**Additional context**
Will allow for easy installation when it comes to controlling the power of the computer, no need to open up the case and add wires between the PiKVM and motherboard.

**Comment by bwildenhain** [comment]
Did you consider using one of the plugs supported by Tasmota (see https://templates.blakadder.com/plug.html)? They are common in home automation. Disadvantage: Usually they can only be connected using wifi (which is not optimal from IT security perspective, but is safer in electric safety perspective as you don't connect your KVM to something in contact with mains power). Advantage: Many of them are quite cheap and even offer power measurement.

**Comment by jidckii** [comment]
I agree — it would be extremely convenient to have everything in one device, maybe a module with several outlets.

Right now, I bought an **ATEN KN1000A** for the company, since I found out about the **PiKVM** project too late.
But the ATEN device covers 100% of the required functions — it has a serial console, KVM, and power control.
At the moment, it looks like this:

<img width="80%" alt="Image" src="https://github.com/user-attachments/assets/80fae4c5-c794-4dc8-8923-71b6ec231ed1" />

<img width="80%" alt="Foto" src="https://github.com/user-attachments/assets/74d25b50-9813-4934-9148-8c1f47a010f7" />

---

Most smart sockets work over Wi-Fi and through private clouds, which are not always accessible in server rooms.
The only more or less vendor-lock-free option I’ve found is this socket:
[https://plcontroller.ru/product/plcontroller-wi-fi-http-smart-socket-pdu-2/](https://plcontroller.ru/product/plcontroller-wi-fi-http-smart-socket-pdu-2/)

<img width="30%" alt="Image" src="https://github.com/user-attachments/assets/1c17e8d9-d4f5-4a55-8bec-ef2e4178a809" />

---

So, in my opinion, this request is more than relevant.
